driver-core/src/test/unit/com/mongodb/MongoClientSettingsSpecification.groovy

```diff
@@ -218,6 +222,21 @@ class MongoClientSettingsSpecification extends Specification {
         thrown(IllegalArgumentException)
     }
 
+    def 'should throw an exception if the timeout is set and negative'() {
+        when:
+        MongoClientSettings.builder().timeout(-1, TimeUnit.SECONDS).build()
+
+        then:
+        thrown(IllegalArgumentException)
+
+        when:
+        def connectionString = new ConnectionString('mongodb://localhost/?timeoutMS=-1')
+        MongoClientSettings.builder().applyConnectionString(connectionString).build()
+
+        then:
+        thrown(IllegalStateException)
```

Comment:
Currently we don't validate connection string options until they are converted into settings. So rather I opted to use the `isTrue` helper rather than `isTrueArgument`.  This is why this route produces an IlleagalStateException.

Happy to convert to an IllegalArgumentException.